ui/packages/app/src/ApiDocs.svelte

```diff
@@ -12,130 +32,282 @@
+					</h4>
 					<div
-						class="bg-gray-50 border border-gray-100 dark:bg-gray-800 dark:border-gray-700 p-6 rounded"
+						class="block bg-white border dark:bg-gray-800 p-4 font-mono text-sm rounded-lg"
 					>
-						<h3 class="text-3xl text-amber-500 font-semibold mb-2">
-							POST /run/{dependency.api_name}
-						</h3>
-						<div class="mb-6">
-							Full URL: <span class="underline"
-								>{root}run/{dependency.api_name}</span
+						&#123;<br />
+						&nbsp;&nbsp;"data": [<br />
+						{#each dependency.inputs as component_id, component_index}
+							&nbsp;&nbsp;&nbsp;&nbsp;<input
+								class="bg-gray-100 dark:bg-gray-600 border-none w-40 px-1 py-0.5 my-0.5 text-sm rounded ring-1 ring-gray-300 dark:ring-gray-500"
+								type="text"
+								bind:value={dependency_inputs[dependency_index][
+									component_index
+								]}
+							/>
+							<span class="text-gray-500">
+								: {instance_map[component_id].documentation?.input_type},</span
 							>
-							<button
-								class="ml-1 px-2 py-0.5 rounded bg-gray-200 dark:bg-gray-700"
-								on:click={() => {
-									navigator.clipboard.writeText(
-										root + "run/" + dependency.api_name
-									);
-									just_copied = d;
-									setTimeout(() => {
-										just_copied = -1;
-									}, 500);
-								}}
-							>
-								{#if just_copied === d}copied!{:else}copy{/if}
-							</button>
-						</div>
-						<h4 class="text-2xl mt-6 mb-4">Input Payload</h4>
-						<div
-							class="block mb-4 text-lg bg-gray-100 dark:bg-gray-700 p-4 font-mono"
-						>
-							&#123;<br />
-							&nbsp;&nbsp;"data": [<br />
-							{#each dependency.documentation[0] as dependency_doc, i}
-								&nbsp;&nbsp;&nbsp;&nbsp;{dependency_doc[1]},
-								<span class="text-pink-400 dark:text-pink-600"
-									>// represents {dependency_doc[0]} of
-									{((label) => {
-										return label ? "'" + label + "'" : "the";
-									})(
-										components.filter((c) => c.id === dependency.inputs[i])[0]
-											.props.label
-									)}
+							<span class="text-gray-400"
+								>// represents {instance_map[component_id].documentation
+									?.input_description} of
+								{((label) => {
+									return label ? "'" + label + "'" : "the";
+								})(instance_map[component_id].props.label)}
 
-									<span class="capitalize"
-										>{components.filter((c) => c.id === dependency.inputs[i])[0]
-											.props.name}</span
-									> component
-								</span>
-								<br />
-							{/each}
-							&nbsp;&nbsp;]<br />
-							&#125;
-						</div>
-						<h4 class="text-2xl mt-6 mb-4">Response Object</h4>
+								<span class="capitalize"
+									>{instance_map[component_id].props.name}</span
+								> component
+							</span>
+							<br />
+						{/each}
+						&nbsp;&nbsp;]<br />
+						&#125;
+					</div>
+					<button
+						on:click={run.bind(null, dependency_index)}
+						class="gr-button gr-button-lg gr-button-primary w-full mt-4"
```

Comment:
We could add a clear/reset button.
<img width="1266" alt="image" src="https://user-images.githubusercontent.com/3841370/201987728-37006c2f-6ed6-479d-9e68-e0c0208e88d6.png">

Reply:
haha I feel like at this point we're just recreating gradio in the api page :P